elemental-operator is silent if it cannot find the local machineRegistration file

If you run `elemental-operator register /path/without/yaml` all you get in the log is

```
Operator version 0.3.0, commit 90791e4, commit date git20220720
```
and it exits.

I would expect a warning (error ?) saying something like

```
Warning: No userdata.yaml found at /path/without/yaml
```

Running `elemental-operator register /non/existant/path` results in a memfault

```
# elemental-operator register /foo/bar
INFO[0000] Operator version 0.3.0, commit 90791e4, commit date git20220720 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x181fef2]

goroutine 1 [running]:
github.com/rancher/elemental-operator/cmd/operator/register.NewRegisterCommand.func1.1({0x7fff9a8b7a6a, 0x8}, {0x0, 0x0}, {0x19a1c20?, 0x1fdcb50?})
	/home/abuild/rpmbuild/BUILD/elemental-operator-0.3.0+git20220720.90791e4/cmd/operator/register/root.go:68 +0x32
path/filepath.WalkDir({0x7fff9a8b7a6a, 0x8}, 0x1e345d0)
	/usr/lib64/go/1.18/src/path/filepath/path.go:479 +0x50
github.com/rancher/elemental-operator/cmd/operator/register.NewRegisterCommand.func1(0xc00023a000?, {0xc0003ef9a0, 0x1, 0x1})
	/home/abuild/rpmbuild/BUILD/elemental-operator-0.3.0+git20220720.90791e4/cmd/operator/register/root.go:67 +0x45b
github.com/spf13/cobra.(*Command).execute(0xc00023a000, {0xc0003ef970, 0x1, 0x1})
	/home/abuild/rpmbuild/BUILD/elemental-operator-0.3.0+git20220720.90791e4/vendor/github.com/spf13/cobra/command.go:876 +0x67b
github.com/spf13/cobra.(*Command).ExecuteC(0xc0001e1900)
	/home/abuild/rpmbuild/BUILD/elemental-operator-0.3.0+git20220720.90791e4/vendor/github.com/spf13/cobra/command.go:990 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	/home/abuild/rpmbuild/BUILD/elemental-operator-0.3.0+git20220720.90791e4/vendor/github.com/spf13/cobra/command.go:918
main.main()
	/home/abuild/rpmbuild/BUILD/elemental-operator-0.3.0+git20220720.90791e4/cmd/operator/main.go:38 +0xaf
```

--- a/cmd/operator/register/root.go
+++ b/cmd/operator/register/root.go
@@ -31,6 +31,7 @@ import (
 	"github.com/rancher/elemental-operator/pkg/tpm"
 	"github.com/rancher/elemental-operator/pkg/version"
 	agent "github.com/rancher/system-agent/pkg/config"
+	"github.com/sanity-io/litter"
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
@@ -69,6 +70,13 @@ func NewRegisterCommand() *cobra.Command {
 			}
 
 			for _, arg := range args {
+				_, err := os.Stat(arg)
+				if err != nil {
+					logrus.Warnf("cannot access config path %s: %s", arg, err.Error())
+					continue
+				} else {
+					logrus.Debugf("scanning config path %s", arg)
+				}
 				viper.AddConfigPath(arg)
 				_ = filepath.WalkDir(arg, func(path string, d fs.DirEntry, err error) error {
 					if err != nil {
@@ -80,6 +88,7 @@ func NewRegisterCommand() *cobra.Command {
 						if err := viper.MergeInConfig(); err != nil {
 							logrus.Fatalf("failed to read config %s: %s", path, err)
 						}
+						logrus.Infof("reading config file %s", path)
 					}
 					return nil
 				})
@@ -99,6 +108,9 @@ func NewRegisterCommand() *cobra.Command {
 					config.Elemental.Registration.Labels[parts[0]] = parts[1]
 				}
 			}
+
+			logrus.Debugf("input config:\n%s", litter.Sdump(config))
+
 			run(config)
 		},
 	}
@@ -134,6 +146,9 @@ func run(config cfg.Config) {
 			logrus.Error(err)
 		}
 	} else {
+		if registration.CACert == "" {
+			logrus.Warn("CACert is empty")
+		}
 		caCert = []byte(registration.CACert)
 	}
 

--- a/go.mod
+++ b/go.mod
@@ -64,12 +64,6 @@ require (
 )
 
 require (
-	github.com/pin/tftp v2.1.0+incompatible // indirect
-	github.com/rancher/aks-operator v1.0.2 // indirect
-	github.com/rancher/eks-operator v1.1.1 // indirect
-	github.com/rancher/gke-operator v1.1.1 // indirect
-	github.com/rancher/norman v0.0.0-20210608202517-59b3523c3133 // indirect
-	github.com/rancher/rke v1.3.1 // indirect
 	github.com/beorn7/perks v1.0.1 // indirect
 	github.com/blang/semver v3.5.1+incompatible // indirect
 	github.com/cespare/xxhash/v2 v2.1.2 // indirect
@@ -118,11 +112,17 @@ require (
 	github.com/opencontainers/image-spec v1.0.2 // indirect
 	github.com/pelletier/go-toml v1.9.4 // indirect
 	github.com/pierrec/lz4 v2.6.1+incompatible // indirect
+	github.com/pin/tftp v2.1.0+incompatible // indirect
 	github.com/prometheus/client_golang v1.12.1 // indirect
 	github.com/prometheus/client_model v0.2.0 // indirect
 	github.com/prometheus/common v0.34.0 // indirect
 	github.com/prometheus/procfs v0.7.3 // indirect
+	github.com/rancher/aks-operator v1.0.2 // indirect
+	github.com/rancher/eks-operator v1.1.1 // indirect
+	github.com/rancher/gke-operator v1.1.1 // indirect
+	github.com/rancher/norman v0.0.0-20210608202517-59b3523c3133 // indirect
 	github.com/rancher/remotedialer v0.2.6-0.20220104192242-f3837f8d649a // indirect
+	github.com/rancher/rke v1.3.1 // indirect
 	github.com/rancher/wharfie v0.4.3 // indirect
 	github.com/russross/blackfriday/v2 v2.1.0 // indirect
 	github.com/spf13/afero v1.6.0 // indirect
@@ -181,3 +181,5 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.0 // indirect
 	sigs.k8s.io/yaml v1.3.0
 )
+
+require github.com/sanity-io/litter v1.5.5 // indirect

